thriftbp/errors.go

```diff
@@ -59,3 +59,13 @@ func BaseplateErrorFilter(codes ...int32) retrybp.Filter {
 		return next(err)
 	}
 }
+
+// NewBaseplateError is a helper function for creating baseplate.Error thrift objects.
+func NewBaseplateError(code baseplatethrift.ErrorCode, message string) *baseplatethrift.Error {
+	asInt32 := int32(code)
+	return &baseplatethrift.Error{
+		Code:    &asInt32,
+		Message: &message,
+		Details: make(map[string]string),
```

Comment:
I think this behavior should be documented, e.g. we don't expect caller to return this error directly, we expect them to fill in the details into the error returned then return the error.